goldmane/proto/methods.go

```diff
@@ -20,26 +20,55 @@ import (
 	"strings"
 
 	"github.com/sirupsen/logrus"
+	"k8s.io/kubernetes/pkg/apis/core/validation"
 )
 
 // ToString converts a PolicyHit struct into a string label.
 // TODO: This is a temporary solution - we should be pushing the structured PolicyHit representation
 // further down the stack, rather than converting between string / struct.
-func (h *PolicyHit) ToString() string {
+func (h *PolicyHit) ToString() (string, error) {
 	// Format is: "<policy index>|<tier>|<namePart>|<action>|<rule index>"
 	tmpl := "%d|%s|%s|%s|%d"
 
+	if err := h.Validate(); err != nil {
+		logrus.WithFields(h.fields()).WithError(err).Error("Failed to validate policy hit")
+		return "", err
+	}
+
 	var namePart string
+	var err error
 	if h.Kind == PolicyKind_EndOfTier {
-		namePart = makeNamePart(h.Trigger)
+		namePart, err = makeNamePart(h.Trigger)
 	} else {
-		namePart = makeNamePart(h)
+		namePart, err = makeNamePart(h)
+	}
+	if err != nil {
+		logrus.WithFields(h.fields()).WithError(err).Error("Failed to generate name part")
+		return "", err
 	}
 
 	// Convert action from enum to string.
 	action := strings.ToLower(Action_name[int32(h.Action)])
 
-	return fmt.Sprintf(tmpl, h.PolicyIndex, h.Tier, namePart, action, h.RuleIndex)
+	return fmt.Sprintf(tmpl, h.PolicyIndex, h.Tier, namePart, action, h.RuleIndex), nil
+}
+
+func (h *PolicyHit) Validate() error {
+	if _, ok := Action_name[int32(h.Action)]; !ok {
+		return fmt.Errorf("unexpected action: %v", h.Action)
+	}
+
+	switch h.Kind {
+	case PolicyKind_GlobalNetworkPolicy, PolicyKind_AdminNetworkPolicy, PolicyKind_BaselineAdminNetworkPolicy:
```

Comment:
It looks like we're missing `PolicyKind_StagedGlobalNetworkPolicy` from this case